internal/common/common_test.go

```diff
@@ -151,25 +150,35 @@ func TestParseReferenceValues(t *testing.T) {
 
 func TestFromProvenance_Amber(t *testing.T) {
 	path := filepath.Join(testdataPath, provenanceExamplePath)
-	provenance, err := amber.ParseProvenanceFile(path)
+	got, err := amber.ParseProvenanceFile(path)
 	if err != nil {
 		t.Fatalf("couldn't parse the provenance file: %v", err)
 	}
 
-	want := NewProvenanceIR("322527c0260e25f0e9a2595bd0d71a52294fe2397a7af76165190fd98de8920d",
-		WithBuildType("https://github.com/project-oak/transparent-release/schema/amber-slsa-buildtype/v1/provenance.json"),
-		WithBuildCmd([]string{"cp", "testdata/static.txt", "test.txt"}),
-		WithBuilderImageSHA256Digest("9e2ba52487d945504d250de186cb4fe2e3ba023ed2921dd6ac8b97ed43e76af9"),
-		WithRepoURIs([]string{"https://github.com/project-oak/transparent-release"}))
+	testutil.AssertEq(t, "binary digest", got.GetBinarySHA256Digest(), "322527c0260e25f0e9a2595bd0d71a52294fe2397a7af76165190fd98de8920d")
+
+	gotBuildType, err := got.GetBuildType()
+	if err != nil {
+		t.Fatalf("couldn't get build type")
+	}
+	testutil.AssertEq(t, "build type", gotBuildType, amber.AmberBuildTypeV1)
 
-	got, err := FromProvenance(provenance)
+	gotBuildCmd, err := got.GetBuildCmd()
 	if err != nil {
-		t.Fatalf("couldn't map provenance to ProvenanceIR: %v", err)
+		t.Fatalf("couldn't get build cmd")
 	}
+	testutil.AssertEq(t, "build cmd[0]", gotBuildCmd[0], "cp")
+	testutil.AssertEq(t, "build cmd[1]", gotBuildCmd[1], "testdata/static.txt")
+	testutil.AssertEq(t, "build cmd[2]", gotBuildCmd[2], "test.txt")
 
-	if diff := cmp.Diff(got, want, cmp.AllowUnexported(ProvenanceIR{})); diff != "" {
```

Comment:
I had intentionally used `cmp.Diff`. This is the standard pattern that we should use everywhere (see #188). Please revert this change, and rewrite the test using `cmp`.

Reply:
Actually, what exactly is this test case testing? The name still says `TestFromProvenance_Amber`, but `FromProvenance` is replaced with `SetProvenanceData`, which is not even called in this test.

I think fixing this test will be easier once `SetProvenanceData` is fixed (ref my other comments on `SetXYZData`).